src/SignedPetriNets.jl

```diff
@@ -0,0 +1,60 @@
+module SignedPetriNets
+export AbstractSignedPetriNet, SignedPetriNet, OpenSignedPetriNet, OpenSignedPetriNetOb,
+  AbstractLabelledSignedPetriNet, LabelledSignedPetriNet, OpenLabelledSignedPetriNet, OpenLabelledSignedPetriNetOb
+
+using AlgebraicPetri
+using Catlab, Catlab.CategoricalAlgebra
+
+import AlgebraicPetri: Open
+
+@present SchSignedPetriNet <: SchPetriNet begin
+  L::Ob
+  Z2::AttrType
+
+  ls::Hom(L, S)
+  lt::Hom(L, T)
+
+  sign::Attr(L, Z2)
+end
+
+@abstract_acset_type AbstractSignedPetriNet <: AbstractPetriNet
+@acset_type SignedPetriNetUntyped(SchSignedPetriNet, index=[:it, :is, :ot, :os]) <: AbstractSignedPetriNet
+const SignedPetriNet = SignedPetriNetUntyped{Bool}
+const OpenSignedPetriNetObUntyped, OpenSignedPetriNetUntyped = OpenACSetTypes(SignedPetriNetUntyped, :S)
+const OpenSignedPetriNetOb, OpenSignedPetriNet = OpenSignedPetriNetObUntyped{Bool}, OpenSignedPetriNetUntyped{Bool}
+
+Open(p::AbstractSignedPetriNet) = OpenSignedPetriNet(p, map(x -> FinFunction([x], ns(p)), 1:ns(p))...)
+Open(p::AbstractSignedPetriNet, legs...) = OpenSignedPetriNet(p, map(l -> FinFunction(l, ns(p)), legs)...)
+Open(n, p::AbstractSignedPetriNet, m) = Open(p, n, m)
+
+function (::Type{T})(pn::AbstractPetriNet, signs::Vararg{Union{Pair,Tuple}}) where T <: AbstractSignedPetriNet
```

Comment:
We should also support `signs <: AbstractVector` here so that you don't have to splat a list longer than 32 elements.

Reply:
Also needs a docstring documenting the structure of the signs argument. It looks like a list of `(s, (t, +/-1))` from the for loop that is using them.

Reply:
For the abstract vector the elements are the same structure that you are currently expecting it's just that with the Vararg you are saying that if a user has a list that they want to pass in, they have to splat that list with ... . You want to also have a type signature that does the same thing if they pass the list in directly. Because splatting is inefficient for long lists.

Reply:
oh I see I see, yeah I can add that!

Reply:
revised my commit to do this.